src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -448,7 +451,8 @@ public int getNumShards(double localDensity)
             {
```

Comment:
Could you also change the lines above to 
```
            // Minimum size only applies if the base count would result in smaller sstables.
            // We also want to use the min size if we don't yet know the flush size (density is NaN).
            // Note: the minimum size cannot be larger than the target size's minimum.
            if (!(count >= baseShardCount)) // also true for count == NaN
```
to make sure we use 1 shard when we don't yet know the size? Some of the tests will need to be updated to reflect that change.

Reply:
OSS C* will get this change as part of CASSANDRA-18753.